lib/m2ts/caption-stream.js

```diff
@@ -319,77 +494,162 @@ var Cea608Stream = function() {
+        // Then byte-shift to the left and OR with char1 so we can pass the
+        // entire character code to `getCharFromCode`.
+        char0 = (char0 & 0x03) << 8;
+        text = getCharFromCode(char0 | char1);
+        this.column_++;
+
+      // Handle extended characters
+      } else if ((char0 === (this.ext_ + 1) || char0 === (this.ext_ + 2)) &&
+          (char1 >= 0x20 && char1 <= 0x3f)) {
+        // Extended characters always follow their "non-extended" equivalents.
+        // IE if a "è" is desired, you'll always receive "eè"; non-compliant
+        // decoders are supposed to drop the "è", while compliant decoders
+        // backspace the "e" and insert "è".
+
+        // Delete the previous character
+        if (this.mode_ === 'popOn') {
+          this.nonDisplayed_[this.row_] = this.nonDisplayed_[this.row_].slice(0, -1);
+        } else {
+          this.displayed_[BOTTOM_ROW] = this.displayed_[BOTTOM_ROW].slice(0, -1);
+        }
+
+        // Bitmask char0 so that we can apply character transformations
+        // regardless of field and data channel.
+        // Then byte-shift to the left and OR with char1 so we can pass the
+        // entire character code to `getCharFromCode`.
+        char0 = (char0 & 0x03) << 8;
+        text = getCharFromCode(char0 | char1);
+        this.column_++;
+
+      // Detect mid-row codes
+      } else if (char0 === this.ext_ && (char1 >= 0x20 && char1 <= 0x2f)) {
+        // According to the standard, mid-row codes
+        // should be replaced with spaces, so add one now
+        text += ' ';
+        this.column_++;
+
+        // Attributes are not additive, so clear all formatting
+        this.clearFormatting(packet.pts);
+
+        if ((char1 & 0xe) === 0xe) {
+          this.addFormatting(packet.pts, 'i');
+        }
+
+        if ((char1 & 0x1) === 0x1) {
+          this.addFormatting(packet.pts, 'u');
+        }
+
+      // Detect offset control codes, but don't affect output
+      } else if (char0 === this.offset_ && (char1 >= 0x21 && char1 <= 0x23)) {
+        this.column_ += (char1 & 0x03);
+        return; // No text will be added to the caption, so just return now
+
+      // Detect PACs (Preamble Address Codes)
+      } else if (char0 >= this.base_ && char0 < (this.base_ + 8)) {
+
+        // There's no logic for PAC -> row mapping, so we have to just
+        // iterate through an array and compare :(
+        ROWS.forEach(function(baseCode, index) {
+          if ((data & 0x1f20) === baseCode) {
+            if (index !== this.row_) {
+              // formatting is only persistent for current row
+              this.clearFormatting(packet.pts);
+              this.row_ = index;
+            }
+
+            if ((data & 0x10) === 0x10) {
+              // we've got an indent level
+              this.column_ = ((data & 0xe) >> 1) * 4;
+            }
+
+            return; // we've found the row so break out of this function
```

Comment:
I guess you mean due to your above-mentioned suggestion? Because, unless JavaScript has somehow gained omniscient powers, I don't know how it would know to break out of the for loop before it has gone through all items?